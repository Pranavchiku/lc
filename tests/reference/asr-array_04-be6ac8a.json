{
    "basename": "asr-array_04-be6ac8a",
    "cmd": "lc --asr-dump --no-color {infile} -o {outfile} -extra-arg=\"\"",
    "infile": "tests/../integration_tests/array_04.cpp",
    "infile_hash": "a3e79a9500684211e6e080dc262e6f1e5005e257df6d05381f9dc9c6",
    "outfile": null,
    "outfile_hash": null,
    "stdout": "asr-array_04-be6ac8a.stdout",
    "stdout_hash": "8ccda65df8bedf43f3a0c20870b16ce84c05bc2dad0df3e41ddd122e",
    "stderr": null,
    "stderr_hash": null,
    "returncode": 0
}