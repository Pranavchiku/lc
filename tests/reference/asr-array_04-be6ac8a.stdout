(TranslationUnit
    (SymbolTable
        1
        {
            main:
                (Function
                    (SymbolTable
                        2
                        {
                            __return_var:
                                (Variable
                                    2
                                    __return_var
                                    []
                                    ReturnVar
                                    ()
                                    ()
                                    Default
                                    (Integer 4)
                                    ()
                                    Source
                                    Public
                                    Required
                                    .false.
                                ),
                            arr1:
                                (Variable
                                    2
                                    arr1
                                    []
                                    Local
                                    ()
                                    ()
                                    Default
                                    (Allocatable
                                        (Array
                                            (Real 8)
                                            [(()
                                            ())
                                            (()
                                            ())]
                                            DescriptorArray
                                        )
                                    )
                                    ()
                                    Source
                                    Public
                                    Required
                                    .false.
                                ),
                            arr2:
                                (Variable
                                    2
                                    arr2
                                    []
                                    Local
                                    ()
                                    ()
                                    Default
                                    (Allocatable
                                        (Array
                                            (Real 8)
                                            [(()
                                            ())]
                                            DescriptorArray
                                        )
                                    )
                                    ()
                                    Source
                                    Public
                                    Required
                                    .false.
                                )
                        })
                    main
                    (FunctionType
                        []
                        (Integer 4)
                        Source
                        Implementation
                        ()
                        .false.
                        .false.
                        .false.
                        .false.
                        .false.
                        []
                        .false.
                    )
                    []
                    []
                    [(Allocate
                        [((Var 2 arr1)
                        [((IntegerConstant 0 (Integer 4))
                        (IntegerConstant 3 (Integer 4)))
                        ((IntegerConstant 0 (Integer 4))
                        (IntegerConstant 3 (Integer 4)))]
                        ()
                        ())]
                        ()
                        ()
                        ()
                    )
                    (=
                        (Var 2 arr1)
                        (ArrayReshape
                            (ArrayConstant
                                [(RealConstant
                                    1.000000
                                    (Real 8)
                                )
                                (RealConstant
                                    2.000000
                                    (Real 8)
                                )
                                (RealConstant
                                    3.000000
                                    (Real 8)
                                )
                                (RealConstant
                                    2.000000
                                    (Real 8)
                                )
                                (RealConstant
                                    5.000000
                                    (Real 8)
                                )
                                (RealConstant
                                    7.000000
                                    (Real 8)
                                )
                                (RealConstant
                                    2.000000
                                    (Real 8)
                                )
                                (RealConstant
                                    5.000000
                                    (Real 8)
                                )
                                (RealConstant
                                    7.000000
                                    (Real 8)
                                )]
                                (Array
                                    (Real 8)
                                    [((IntegerConstant 0 (Integer 4))
                                    (IntegerConstant 9 (Integer 4)))]
                                    FixedSizeArray
                                )
                                RowMajor
                            )
                            (ArrayConstant
                                [(ArraySize
                                    (Var 2 arr1)
                                    (IntegerConstant 1 (Integer 4))
                                    (Integer 4)
                                    ()
                                )
                                (ArraySize
                                    (Var 2 arr1)
                                    (IntegerConstant 2 (Integer 4))
                                    (Integer 4)
                                    ()
                                )]
                                (Array
                                    (Integer 4)
                                    [((IntegerConstant 0 (Integer 4))
                                    (IntegerConstant 2 (Integer 4)))]
                                    FixedSizeArray
                                )
                                RowMajor
                            )
                            (Allocatable
                                (Array
                                    (Real 8)
                                    [(()
                                    ())
                                    (()
                                    ())]
                                    DescriptorArray
                                )
                            )
                            ()
                        )
                        ()
                    )
                    (Allocate
                        [((Var 2 arr2)
                        [((IntegerConstant 0 (Integer 4))
                        (IntegerConstant 3 (Integer 4)))]
                        ()
                        ())]
                        ()
                        ()
                        ()
                    )
                    (=
                        (Var 2 arr2)
                        (ArrayConstant
                            [(RealConstant
                                5.000000
                                (Real 8)
                            )
                            (RealConstant
                                6.000000
                                (Real 8)
                            )
                            (RealConstant
                                7.000000
                                (Real 8)
                            )]
                            (Array
                                (Real 8)
                                [((IntegerConstant 0 (Integer 4))
                                (IntegerConstant 3 (Integer 4)))]
                                FixedSizeArray
                            )
                            RowMajor
                        )
                        ()
                    )
                    (Print
                        [(Var 2 arr2)]
                        ()
                        ()
                    )
                    (=
                        (Var 2 __return_var)
                        (IntegerConstant 0 (Integer 4))
                        ()
                    )
                    (Return)]
                    (Var 2 __return_var)
                    Public
                    .false.
                    .false.
                    ()
                ),
            main_program:
                (Program
                    (SymbolTable
                        3
                        {
                            exit_code:
                                (Variable
                                    3
                                    exit_code
                                    []
                                    Local
                                    ()
                                    ()
                                    Default
                                    (Integer 4)
                                    ()
                                    Source
                                    Public
                                    Required
                                    .false.
                                )
                        })
                    main_program
                    [main]
                    [(=
                        (Var 3 exit_code)
                        (FunctionCall
                            1 main
                            1 main
                            []
                            (Integer 4)
                            ()
                            ()
                        )
                        ()
                    )]
                )
        })
    []
)
